### ISSUE TITLE
Provide esm version

### ISSUE THREAD

**Issue by tobiasdiez** [redis/node-redis#2434]
### Motivation

With more and more build tools (vite, nuxt, etc) going towards an esm-first build, it would be nice if a proper [esm interface](https://nodejs.org/api/esm.html) could be provided, i.e. transpile and distribute esm version, including correct [export](https://nodejs.org/api/packages.html#conditional-exports) declaration in `package.json`.

**Comment by leibale** [comment]
@tobiasdiez is there a way to do it without doubling the "Unpacked Size"?

**Comment by tobiasdiez** [comment]
No not really, you do need to distribute both versions. But the metric "unpacked size" is misleading anyway, since distributing an esm version would actually allow bundlers to treeshake effectively so that the effective size for peoples apps decreases.

**Comment by leibale** [comment]
@tobiasdiez I assume most people do not bundle the server code, let alone tree-shaking.. maybe in this new world of "edge functions" I'm incorrect..?

**Comment by tobiasdiez** [comment]
Yes, reducing the distribution size is particularly important for edge/serverless, but more generally many popular fullstack frameworks like next or nuxt run the server build through webpack or vite, and they would profit from esm support.

**Comment by jay-l-e-e** [comment]
I propose [Solidis](https://github.com/vcms-io/solidis) as an alternative solution to this problem.

**Comment by nekomeowww** [comment]
Any plan for this?
